invoke-ai/InvokeAI
# Added updater functionality

## Summary

- Added functionality to check for new available versions and alert the user to them
- Added functionality to download and run the latest version of the InvokeAI Installer, which brings back the ability to update InvokeAI directly from the launcher.
- Foundation is in place to support listing additions, changes, and warnings on the update screen so the user can see them without needing to access the Github page.
- **Note: When the actual downloaded installer is run, it will list the wrong file path for the user's invokeai directory. This seems to stem from an issue with the python file involved with the installer. It needs to generally point to a relative path of `~/invokeai` instead of trying to follow the file path the installer is located in.**

## Related Issues / Discussions

This re-adds the built-in updating functionality that was lost with version 4.0+, but now avoids the issues of `pip` trying to self-update itself.

## QA Instructions

The updater functionality has been added in as option 3 on the launch menu. Test it out, either for updating or reinstalling, and see if everything works as expected.

## Merge Plan

This should be clear to merge without much hassle overall, since only the `invoke.sh` file is involved.

## Checklist
